sagemaker/launch/launch_gconstruct.py

```diff
@@ -95,44 +95,38 @@ def run_job(input_args, image, unknowargs):
     )
 
 
-def parse_args():
-    parser = argparse.ArgumentParser()
-
-    parser.add_argument("--image-url", type=str,
-        help="Processing docker image")
-    parser.add_argument("--role", type=str,
-        help="SageMaker role")
-    parser.add_argument("--instance-type", type=str,
-        default=INSTANCE_TYPE,
-        help="instance type used to train models")
-    parser.add_argument("--region", type=str,
-        default="us-east-1",
-        help="Region")
-    parser.add_argument("--task-name", type=str,
-        default=None, help="User defined SageMaker task name")
-    parser.add_argument("--input-graph-s3", type=str,
-        default=None, help="S3 location store the input graph data")
-    parser.add_argument("--output-graph-s3", type=str,
-        default=None, help="S3 location store the constructed graph")
-    parser.add_argument("--volume-size-in-gb", type=int,
-        default=INSTANCE_VOLUME_SIZE,
-        help="Size in GB of the EBS volume to use for storing data during processing")
-    parser.add_argument("--entry-point", type=str,
+def get_gconstruct_parser():
+    """
+    Get GraphStorm GConstruct task parser.
+    """
+    parser = get_common_parser()
+
+    gconstruct_arguments = parser.add_argument_group("GraphStorm gconstruct arguments")
+
+    gconstruct_arguments.add_argument("--input-graph-s3", type=str,
+        required=True, help="S3 location of the input graph data")
+    gconstruct_arguments.add_argument("--output-graph-s3", type=str,
+        required=True, help="S3 location to store the constructed graph")
+    gconstruct_arguments.add_argument("--entry-point", type=str,
         default="graphstorm/sagemaker/run/gconstruct_entry.py",
         help="PATH-TO graphstorm/sagemaker/run/gconstruct_entry.py")
 
-    parser.add_argument("--graph-name", type=str,
+    gconstruct_arguments.add_argument("--graph-name", type=str,
         required=True, help="Graph name")
-    parser.add_argument("--graph-config-file", type=str,
-        required=True, help="Graph configuration file. It must be a relative path to --input-graph-s3.")
+    gconstruct_arguments.add_argument("--graph-config-file", type=str,
+        required=True, help="Graph configuration file. It must be a relative "
+        "S3 path to the --input-graph-s3 prefix.")
 
     return parser
 
 if __name__ == "__main__":
-    arg_parser = parse_args()
+    arg_parser = get_gconstruct_parser()
     args, unknownargs = arg_parser.parse_known_args()
     print(args)
 
     gconstruct_image = args.image_url
 
+    if not args.instance_type:
```

Comment:
In common_parser L23 we force users to provide an instance_type. So this if statement will always be False.

Reply:
Right, I'll remove the `required` from the common parser to allow different default instances for each task.